### ISSUE TITLE
Collision of different Css. In angular page Salesforce

### ISSUE THREAD

**Issue by aby0007** [angular/material#8844]
**Actual behavior**:
-  I am trying to create a simple angular page 
I have
1. Text field 
2. autocomplete. 
3. Rich Text area field. 
4 . An attachment field

When I am trying combine the below different modules.  I am having issues in 
`<labels></labels>`
The label of Rich text area field is sliding down to the customization links of text editor . Unless I start writing something in it. 

http://codepen.io/timbrown81/pen/JKoGMr -  Rich Text Area
http://codepen.io/pen/ - autocomplete

The Css mentioned in 

**CodePen** or Steps to reproduce the issue: *

Just combine the above code from both links and from salesforce perspective I am pasting my code below. 

`<apex:page controller="AutocompleteTestingCls" showHeader="false" readOnly="true">
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    
    
    <script src="//assets.codepen.io/assets/common/stopExecutionOnTimeout.js?t=1"></script>
    <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.5.5/angular.js"></script>
    <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.5.5/angular-animate.min.js"></script>
    <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.5.5/angular-route.min.js"></script>
    <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.5.5/angular-aria.min.js"></script>
    <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.5.5/angular-messages.min.js"></script>
    <script src="https://s3-us-west-2.amazonaws.com/s.cdpn.io/t-114/svg-assets-cache.js"></script>
    <script src="https://cdn.gitcdn.link/cdn/angular/bower-material/v1.1.0-rc.5/angular-material.js"></script>
    
    
    <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700,400italic"/>
    <link rel="stylesheet prefetch" href="https://cdn.gitcdn.link/cdn/angular/bower-material/v1.1.0-rc.5/angular-material.css"/>
    <link rel="stylesheet prefetch" href="https://material.angularjs.org/1.1.0-rc.5/docs.css"/>
    
    
    <link rel="stylesheet prefetch" href="https://ajax.googleapis.com/ajax/libs/angular_material/1.1.0-rc4/angular-material.min.css"/>
    <link rel="stylesheet prefetch" href="https://material.angularjs.org/1.1.0-rc4/docs.css"/>
    <link rel="stylesheet prefetch" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css"/>
    <link rel="stylesheet prefetch" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.6.3/css/font-awesome.css"/>
    <link rel="stylesheet prefetch" href="https://cdnjs.cloudflare.com/ajax/libs/textAngular/1.5.1/textAngular.min.css"/>
    
    
    
    <script src="//assets.codepen.io/assets/common/stopExecutionOnTimeout.js?t=1"></script>
    <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.4.8/angular.js"></script>
    <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.4.8/angular-animate.min.js"></script>
    <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.4.8/angular-route.min.js"></script>
    <script src="https://ajax.googleapis.com/ajax/libs/angularjs/1.4.8/angular-aria.min.js"></script>
    <script src="https://s3-us-west-2.amazonaws.com/s.cdpn.io/t-114/svg-assets-cache.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/angular-sanitize/1.5.6/angular-sanitize.min.js"></script>
    <script src="https://ajax.googleapis.com/ajax/libs/angular_material/1.1.0-rc4/angular-material.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/textAngular/1.5.0/textAngular.min.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/textAngular/1.5.1/textAngular-sanitize.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/textAngular/1.5.1/textAngular-rangy.min.js"></script>
    
    
    

    
    
    <head>
        <style>
            .chipsdemoContactChips md-content.autocomplete {
                min-height: 250px;
            }
            
            .chipsdemoContactChips md-content.autocomplete input {
                min-width: 400px;
            }
            
            .chipsdemoContactChips .md-item-text.compact {
                padding-top: 8px;
                padding-bottom: 8px;
            }
            
            .chipsdemoContactChips .contact-item {
                box-sizing: border-box;
            }
            
            .chipsdemoContactChips .contact-item.selected {
                opacity: 0.5;
            }
            
            .chipsdemoContactChips .contact-item.selected h3 {
                opacity: 0.5;
            }
            
            .chipsdemoContactChips .contact-item .md-list-item-text {
                padding: 14px 0;
                max-width: 190px;
            }
            
            .chipsdemoContactChips .contact-item .md-list-item-text h3 {
                margin: 0 !important;
                padding: 0;
                line-height: 1.2em !important;
            }
            
            .chipsdemoContactChips .contact-item .md-list-item-text h3,
            .chipsdemoContactChips .contact-item .md-list-item-text p {
                text-overflow: ellipsis;
                white-space: nowrap;
                overflow: hidden;
            }
            
            @media (min-width: 960px) {
                .chipsdemoContactChips .contact-item {
                    float: left;
                    width: 33%;
                }
            }
            
            .chipsdemoContactChips md-contact-chips {
                margin-bottom: 10px;
            }
            
            .chipsdemoContactChips .md-chips {
                padding: 5px 0 8px;
            }
            
            .chipsdemoContactChips .fixedRows {
                height: 250px;
                overflow: hidden;
            }
            .inputdemoErrors .inputErrorsApp {
                min-height: 48px; 
            }
            
            .inputdemoErrors md-input-container > p {
                font-size: 0.8em;
                text-align: left;
                width: 100%; 
             }
        </style>
        
        <script>
            (function () {
                'use strict';
                angular.module('MyApp',['ngMaterial', 'ngMessages', 'material.svgAssetsCache', 'textAngular']).controller('ContactChipDemoCtrl', DemoCtrl);

                function DemoCtrl ($q, $timeout) {
                    var self = this;
                    console.log(self);
                    console.log(this);
                    var pendingSearch, cancelSearch = angular.noop;
                    var cachedQuery, lastSearch;
                    console.log(pendingSearch);
                    console.log(cancelSearch);
                    console.log(cachedQuery);
                    console.log(lastSearch);
                    
                    self.allContacts = loadContacts();
                    console.log(self.allContacts);
                    self.contacts = [self.allContacts[0]];
                    self.asyncContacts = [];
                    self.filterSelected = true;

                    self.querySearch = querySearch;
                    self.delayedQuerySearch = delayedQuerySearch;
                    console.log('self.delayedQuerySearch');
                    console.log(self.delayedQuerySearch);
                    /**
                     * Search for contacts; use a random delay to simulate a remote call
                     */
                    function querySearch (criteria) {
                        cachedQuery = cachedQuery || criteria;
                        return cachedQuery ? self.allContacts.filter(createFilterFor(cachedQuery)) : [];
                    }

                    /**
                     * Async search for contacts
                     * Also debounce the queries; since the md-contact-chips does not support this
                     */
                    function delayedQuerySearch(criteria) {
                      cachedQuery = criteria;
                      if ( !pendingSearch || !debounceSearch() )  {
                        cancelSearch();

                        return pendingSearch = $q(function(resolve, reject) {
                          // Simulate async search... (after debouncing)
                          cancelSearch = reject;
                          $timeout(function() {

                            resolve( self.querySearch() );

                            refreshDebounce();
                          }, Math.random() * 500, true)
                        });
                      }

                      return pendingSearch;
                    }

                    function refreshDebounce() {
                      lastSearch = 0;
                      pendingSearch = null;
                      cancelSearch = angular.noop;
                    }

                    /**
                     * Debounce if querying faster than 300ms
                     */
                    function debounceSearch() {
                      var now = new Date().getMilliseconds();
                      lastSearch = lastSearch || now;

                      return ((now - lastSearch) < 300);
                    }

                    /**
                     * Create filter function for a query string
                     */
                    function createFilterFor(query) {
                      var lowercaseQuery = angular.lowercase(query);

                      return function filterFn(contact) {
                        return (contact._lowername.indexOf(lowercaseQuery) != -1);;
                      };

                    }

                    function loadContacts() {
                        var contacts = [];
                        console.log({!lstUserJ});
                        contacts = {!lstUserJ};
                        
                        //var contacts = ['Marina Augustine','Oddr Sarno','Nick Giannopoulos','Narayana Garner','Anita Gros','Megan Smith','Tsvetko Metzger','Hector Simek','Some-guy withalongalastaname'];
                        return contacts.map(function (c, index) {
                            //var cParts = c.split(' ');
                            console.log(c.Name);
                            console.log(c.email);
                            console.log(c.SmallPhotoUrl);                            
                            var contact = {
                                name: c.Name,
                                email: c.email,
                                image: c.SmallPhotoUrl
                            };
                            contact._lowername = contact.name.toLowerCase();
                            return contact;
                        });
                    }
                }

            })();
        </script>
    </head>

    <div ng-controller="ContactChipDemoCtrl as ctrl" layout="column" ng-cloak="" class="chipsdemoContactChips" ng-app="MyApp">
        
        <md-content class="md-padding autocomplete" layout="column">
            <form name="projectForm">
                
                <md-input-container class="md-block">
                    <label>Subject</label>
                    <input md-maxlength="30" required="true" md-no-asterisk="" name="description" ng-model="project.Subject"/>
                    <div ng-messages="projectForm.description.$error">
                        <div ng-message="required">This is required.</div>
                        <div ng-message="md-maxlength">The name has to be less than 30 characters long.</div>
                    </div>
                </md-input-container>
                
                <h2 class="md-title">User Search</h2>
                <md-contact-chips ng-model="ctrl.asyncContacts" md-contacts="ctrl.delayedQuerySearch($query)" md-contact-name="name" md-contact-image="image" md-contact-email="email" md-require-match="true" md-highlight-flags="i" filter-selected="ctrl.filterSelected" placeholder="To" md-not-found="NOOOOOO">
                    <md-not-found id="Testinggggggggg">
                        No matches found.
                    </md-not-found>
                </md-contact-chips>
                
                
                <md-input-container class="md-block">
                    <label>Description</label>
                    <text-angular ng-model="project.description"></text-angular>
                </md-input-container>
            </form>
        </md-content>
        
    </div>

    
</apex:page>`

----

> Do not modify the titles or questions. Simply add your responses to the ends of the questions.
  Add more lines if needed.

**Comment by ThomasBurleson** [comment]
Please post this in the [Angular Material Forum](https://groups.google.com/forum/#!forum/ngmaterial)
